Node Repave Configuration (#1462)
spectrocloud/librarium: content/docs/04-clusters/06-cluster-management/15-node-pool.md

</WarningBox>

## Repave Behavior and Configuration

In Kubernetes, the term "repave" refers to the process of replacing a node with a new node. [Repaving](/glossary-all#repavement) is a common practice in Kubernetes to ensure that nodes are deployed with the latest version of the operating system and Kubernetes. Repaving is also used to replace nodes that are unhealthy or have failed. You can configure the repave time interval for a node pool. 

> The ability to configure the repave time interval for all node pools except the master pool. The repave time interval is set to 900 seconds (15 minutes) by default. You can configure the node repave time interval during the cluster creation process or after the cluster is created. To modify the repave time interval after the cluster is created, follow the [Change a Node Pool](#changeanodepool) instructions below.

Reviewer: ```suggestion
The ability to configure the repave time interval for all node pools except the master pool. The default repave time interval is 900 seconds (15 minutes). You can configure the node repave time interval during the cluster creation process or after the cluster is created. To modify the repave time interval after the cluster is created, follow the [Change a Node Pool](#changeanodepool) instructions below.
```

| **Availability Zones** | The Availability Zones from which to select available servers for deployment. If you select multiple zones, Palette will deploy servers evenly across them as long as sufficient servers are available to do so. |
| **Disk Size** | Give the required storage size. |
 

<br />

<WarningBox>

Some features may not be available for all infrastructure providers. Review each infrastructure provider's node pool configuration settings to learn more.

</WarningBox>

<br />

## Create a New Node Pool

### Prerequisites

> * A Palette deployed cluster.

Reviewer: ```suggestion
* A Palette-deployed cluster.
```

Reviewer: OR "A cluster deployed in Palette."

* Sufficient permissions to edit the cluster.

### Create Node Pool

> 

Reviewer: Should we add the "Log in to Palette" step as the first step and navigate the user in subsequent steps to the point they click on the Nodes tab?

Author: That's odd, I could swear I had that as the first step. Let me fix that.

<br />

7. Click on **Confirm** to create the new node pool.

### Validate

After you create a new node pool, you can validate the node pool by following the steps below.

1. Log in to [Palette](https://console.spectrocloud.com).

2. Navigate to the left **Main Menu** and click on **Clusters**.

3. Click on the row of the cluster you added the new node pool.

4. Click on the **Nodes** tab.

5. Ensure the new node pool is listed in the **Node Pools** section and that all compute instances are in the healthy status. 

## Change a Node Pool

You can apply changes to a node pool after a cluster is created and deployed. You can change the node pool's taints label, node repavement interval, number of compute instances in the node pool and more. To make changes to an active cluster's node pools, follow the steps below. 

### Prerequisites

* A Palette deployed cluster.

* Sufficient permissions to edit the cluster.

### Edit Node Pool

1. Log in to [Palette](https://console.spectrocloud.com).

2. Navigate to the left **Main Menu** and click on **Clusters**.

> 3. Click on the row of the cluster you want to edit the node pool.

Reviewer: ```suggestion
3. Select a cluster to edit the node pool.
```

4. Click on the **Nodes** tab.

5. The nodes details page is where you can review the existing node pools and their configuration. You can also add a new node pool from this page. Click on the **Edit** button to make changes to the node pool.

6. Make the changes as needed. Refer to the [Node Pool Configuration Settings](#nodepoolconfigurationsettings) tables for more information on each field. 

7. Click on **Confirm** to update the node pool.

### Validate

After you have modified a new node pool, you can validate the node pool by following the steps below.

1. Log in to [Palette](https://console.spectrocloud.com).

2. Navigate to the left **Main Menu** and click on **Clusters**.

> 3. Click on the row of the cluster you added the new node pool.

Reviewer: ```suggestion
3. Select the cluster with the new node pool.
```

Reviewer: OR "Select the cluster to which you added the new node pool."
